homework/home1.1/src/Main.java

```diff
@@ -0,0 +1,51 @@
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        // Input coefficients
+        System.out.print("Enter coefficient a: ");
+        while (!scanner.hasNextDouble()) {
+            System.out.print("Invalid input. Enter coefficient a: ");
+            scanner.next();
+        }
+        double a = scanner.nextDouble();
+
+        if (a == 0) {
+            System.out.println("Coefficient 'a' must not be zero, as this would not be a quadratic equation.");
+            return;
+        }
+
+        System.out.print("Enter coefficient b: ");
+        while (!scanner.hasNextDouble()) {
+            System.out.print("Invalid input. Enter coefficient b: ");
+            scanner.next();
+        }
+        double b = scanner.nextDouble();
+
+        System.out.print("Enter coefficient c: ");
+        while (!scanner.hasNextDouble()) {
+            System.out.print("Invalid input. Enter coefficient c: ");
+            scanner.next();
+        }
+        double c = scanner.nextDouble();
+
+        // Calculate the determinant
+        double determinant = b * b - 4 * a * c;
```

Comment:
encapsulate calculation to a separate function that will perform calculations and return vector of roots. And based on the result size print the solutions. Such an approach allows to split business logic and UI representation of results.